packages/lib/src/utils/Bytecode.ts

```diff
@@ -31,11 +35,33 @@ export function hasUnlinkedVariables(bytecode: string): boolean {
   return getSolidityLibNames(bytecode).length > 0;
 }
 
-// Removes the last 43 bytes of the bytecode, i.e., the swarm hash that the solidity compiler appends and that
-// respects the following structure: 0xa1 0x65 'b' 'z' 'z' 'r' '0' 0x58 0x20 <32 bytes swarm hash> 0x00 0x29
-// (see https://solidity.readthedocs.io/en/v0.4.24/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode)
-export function tryRemoveSwarmHash(bytecode: string): string {
-  return bytecode.replace(/a165627a7a72305820[a-fA-F0-9]{64}0029$/, '');
+// Removes the swarm hash from the CBOR encoded metadata at the end of the bytecode
+// (see https://solidity.readthedocs.io/en/v0.5.9/metadata.html)
+export function tryRemoveMetadata(bytecode: string): string {
+  // Bail on empty bytecode
+  if (!bytecode || bytecode.length <= 2) return bytecode;
+
+  // Gather length of CBOR metadata from the end of the file
+  const rawLength = bytecode.slice(bytecode.length - 4);
+  const length = parseInt(rawLength, 16);
+
+  // Bail on unreasonable values for length (meaning we read something else other than metadata length)
+  if (length > 200) return bytecode;
```

Comment:
Why 200? I would either remove this check or replace it with `length > (bytecode.length - 4)`. The reason is:

> The CBOR mapping can also contain other keys, so it is better to fully decode the data instead of relying on it starting with 0xa265. For example, if any experimental features that affect code generation are used, the mapping will also contain "experimental": true.

CBOR mapping can be way bigger than 200.

Reply:
And if we read something else we'll fail on decoding.

Reply:
> And if we read something else we'll fail on decoding.

Good point!